include/deal.II/numerics/vector_tools.h

```diff
@@ -969,13 +974,14 @@ namespace VectorTools
    * apply as for the previous function, in particular about the use of the
    * component mask and the requires size of the function object.
    */
-  template <typename DoFHandlerType, typename number>
+  template <int dim, int spacedim, template <int,int> class DoFHandlerType,
```

Comment:
You could change this to
``template <int dim, int spacedim, class DoFHandlerType, typename number>``
right?